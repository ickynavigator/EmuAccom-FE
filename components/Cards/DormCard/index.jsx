import {
  Button,
  Card,
  Group,
  Image,
  Text,
  useMantineTheme,
} from "@mantine/core";
import React from "react";

const Index = ({ _id, name, image, description }) => {
  const theme = useMantineTheme();

  const secondaryColor =
    theme.colorScheme === "dark" ? theme.colors.dark[1] : theme.colors.gray[7];

  return (
    <div style={{ width: 340, margin: "auto" }}>
      <Card shadow="sm" p="lg">
        <Card.Section>
          <Image src={image} height={160} alt={name} />
        </Card.Section>

        <Group
          position="apart"
          style={{ marginBottom: 5, marginTop: theme.spacing.sm }}
        >
          <Text weight={500}>{name}</Text>
          {/* <Badge color="pink" variant="light">
            On Sale
          </Badge> */}
        </Group>

        <Text size="sm" style={{ color: secondaryColor, lineHeight: 1.5 }}>
          {description}
        </Text>

        <Button
          variant="light"
          color="blue"
          fullWidth
          style={{ marginTop: 14 }}
          component="a"
          href={`/dorms/${_id}`}
        >
          Check out house now
        </Button>
      </Card>
    </div>
  );
};

export default Index;
