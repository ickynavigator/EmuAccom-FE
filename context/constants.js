// TYPES
export const LOGIN_USER = "LOGIN_USER";
export const LOGOUT_USER = "LOGOUT_USER";
export const AUTHENTICATE_USER = "AUTHENTICATE_USER";
export const LOGIN_MANAGER = "LOGIN_MANAGER";
export const LOGOUT_MANAGER = "LOGOUT_MANAGER";
export const AUTHENTICATE_MANAGER = "AUTHENTICATE_MANAGER";
export const UPDATE_USER = "UPDATE_USER";
