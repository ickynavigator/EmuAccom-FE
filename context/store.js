import { useLocalStorage } from "@mantine/hooks";
import React, { createContext, useEffect, useReducer } from "react";
import combineReducers from "react-combine-reducers";
import { initialState as userState, userReducer } from "./user/reducer";

const appState = {};
const store = createContext(appState);
const { Provider } = store;

const StateProvider = ({ children }) => {
  const [reducer, initialState] = combineReducers({
    user: [userReducer, userState],
  });

  // Fetches the state from the browser local storage
  const [persistedState, setPersistedState] = useLocalStorage({
    key: "state",
    defaultValue: initialState,
  });

  const [state, dispatch] = useReducer(reducer, persistedState);

  useEffect(() => {
    // Updates the local storage state when the user makes changes
    setPersistedState(state);
  }, [setPersistedState, state]);

  return (
    // eslint-disable-next-line react/jsx-filename-extension
    <Provider value={{ state, dispatch }}>{children}</Provider>
  );
};

export { store, StateProvider };
