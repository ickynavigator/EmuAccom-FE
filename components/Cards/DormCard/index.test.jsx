import { ColorSchemeProvider, MantineProvider } from "@mantine/core";
import { render } from "@testing-library/react";
import React from "react";
import { StateProvider } from "../../../context/store";
import DormCard from "./index";

describe("DormCard component tests", () => {
  it("matches the Snapshot", () => {
    const { container } = render(
      <StateProvider>
        <ColorSchemeProvider>
          <MantineProvider>
            <DormCard />
          </MantineProvider>
        </ColorSchemeProvider>
      </StateProvider>,
    );
    expect(container).toMatchSnapshot();
  });
});
