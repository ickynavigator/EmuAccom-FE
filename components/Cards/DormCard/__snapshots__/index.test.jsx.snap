// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DormCard component tests matches the Snapshot 1`] = `
<div>
  <div
    style="width: 340px; margin: auto;"
  >
    <div
      class="mantine-Paper-root mantine-Card-root mantine-tz9aok"
    >
      <div
        class="mantine-Card-cardSection mantine-wudk7j"
      >
        <div
          class="mantine-Image-root mantine-15po0m8"
        >
          <figure
            class="mantine-11nhzn5 mantine-Image-figure"
          >
            <div
              class="mantine-qqmv3w mantine-Image-imageWrapper"
            >
              <img
                class="mantine-19xxpso mantine-Image-image"
                style="object-fit: cover; width: 100%; height: 160px;"
              />
            </div>
          </figure>
        </div>
      </div>
      <div
        class="mantine-Group-root mantine-19jxmdp"
        style="margin-bottom: 5px; margin-top: 12px;"
      >
        <div
          class="mantine-Text-root mantine-Group-child mantine-125basi"
        />
      </div>
      <div
        class="mantine-Text-root mantine-1nlnfit"
        style="color: rgb(73, 80, 87); line-height: 1.5;"
      />
      <a
        class="mantine-Button-light mantine-Button-root mantine-15godsx"
        href="/dorms/undefined"
        style="margin-top: 14px;"
        type="button"
      >
        <div
          class="mantine-3xbgk5 mantine-Button-inner"
        >
          <span
            class="mantine-qo1k2 mantine-Button-label"
          >
            Check out house now
          </span>
        </div>
      </a>
    </div>
  </div>
</div>
`;
